Build step fails kernel version mismatch

When the host kernel version is older than the Fedora repos the module fails to compile.

Host kernel version:
~~~
[bjarvis@bjarvis silverblue-v4l2loopback]$ echo $(uname -r)
5.14.16-301.fc35.x86_64
~~~

Build error (note it fails looking for the modules that match the kernel version):
~~~
[1/2] STEP 8/8: RUN cd /tmp/v4l2loopback-${V4L2LOOPBACK_VERSION};     make -j$(nproc) && make install
Building v4l2-loopback driver...
make -C /lib/modules/`uname -r`/build M=/tmp/v4l2loopback-0.12.5 modules
make[1]: *** /lib/modules/5.14.16-301.fc35.x86_64/build: No such file or directory.  Stop.  <<<---------------------------------
make: *** [Makefile:43: v4l2loopback.ko] Error 2
[2/2] STEP 1/8: FROM registry.fedoraproject.org/fedora:latest
Error: error building at STEP "RUN cd /tmp/v4l2loopback-${V4L2LOOPBACK_VERSION};     make -j$(nproc) && make install": error while running runtime: exit status 2
~~~

Kernel packages installed in container:
~~~
  Installing       : systemd-udev-249.7-2.fc35.x86_64                   123/130 
  Running scriptlet: systemd-udev-249.7-2.fc35.x86_64                   123/130 
  Installing       : libkcapi-1.3.1-3.fc35.x86_64                       124/130 
  Installing       : libkcapi-hmaccalc-1.3.1-3.fc35.x86_64              125/130 
  Installing       : kpartx-0.8.6-5.fc35.x86_64                         126/130 
  Installing       : dracut-055-6.fc35.x86_64                           127/130 
  Installing       : kernel-core-5.15.5-200.fc35.x86_64                 128/130  <<<---------------------------------
  Running scriptlet: kernel-core-5.15.5-200.fc35.x86_64                 128/130 
  Installing       : kernel-modules-5.15.5-200.fc35.x86_64              129/130  <<<---------------------------------
  Running scriptlet: kernel-modules-5.15.5-200.fc35.x86_64              129/130  
  Installing       : kernel-devel-5.15.5-200.fc35.x86_64                130/130  <<<---------------------------------
  Running scriptlet: kernel-devel-5.15.5-200.fc35.x86_64                130/130 
  Running scriptlet: kernel-core-5.15.5-200.fc35.x86_64                 130/130 
~~~

The kernel-* packages that are installed do not match the version on the host system.

--- a/Containerfile
+++ b/Containerfile
@@ -1,15 +1,25 @@
 FROM registry.fedoraproject.org/fedora:latest as builder
 MAINTAINER "Robert Bohne" robert@bohne.io
 
+ARG V4L2LOOPBACK_KERNEL_VERSION
 ARG V4L2LOOPBACK_VERSION
 ARG V4L2LOOPBACK_SHA256
 # 0.12.5 -> e152cd6df6a8add172fb74aca3a9188264823efa5a2317fe960d45880b9406ae
 
 WORKDIR /tmp
-RUN dnf install -y \
+
+# Install koji and use to pull kernel packages based on V4L2LOOPBACK_KERNEL_VERSION
+RUN dnf install -y koji && \
+    mkdir /tmp/koji && \
+    cd /tmp/koji && \
+    koji download-build --arch=x86_64 kernel-${V4L2LOOPBACK_KERNEL_VERSION::-7} 
+
+RUN cd /tmp/koji && \
+    dnf install -y \
     gc gcc glibc-devel glibc-headers \
-    kernel-core kernel-devel kernel-modules && \
+    ./kernel-core-${V4L2LOOPBACK_KERNEL_VERSION}.rpm ./kernel-devel-${V4L2LOOPBACK_KERNEL_VERSION}.rpm ./kernel-modules-${V4L2LOOPBACK_KERNEL_VERSION}.rpm && \
     dnf clean all -y
+    
     
 RUN curl -LS https://github.com/umlaeute/v4l2loopback/archive/v${V4L2LOOPBACK_VERSION}.tar.gz| \
     { t="$(mktemp)"; trap "rm -f '$t'" INT TERM EXIT; cat >| "$t"; sha256sum --quiet -c <<<"${V4L2LOOPBACK_SHA256} $t" \
@@ -26,7 +36,12 @@ ARG V4L2LOOPBACK_VERSION
 ARG V4L2LOOPBACK_KERNEL_VERSION
 WORKDIR /tmp
 
-RUN dnf install -y kernel-core kernel-modules v4l-utils
+COPY --from=builder /tmp/koji/ /tmp/koji/
+
+RUN cd /tmp/koji && \
+    dnf install -y ./kernel-core-${V4L2LOOPBACK_KERNEL_VERSION}.rpm ./kernel-modules-${V4L2LOOPBACK_KERNEL_VERSION}.rpm  v4l-utils && \
+    rm -rf /tmp/koji
+
 COPY --from=builder /tmp/v4l2loopback-${V4L2LOOPBACK_VERSION}/v4l2loopback.ko \
                     /usr/lib/modules/${V4L2LOOPBACK_KERNEL_VERSION}/extra/v4l2loopback.ko
 

